lib/src/Base/Stat/CorrelationAnalysis.cxx

```diff
@@ -26,12 +26,57 @@
 
 BEGIN_NAMESPACE_OPENTURNS
 
-/* Compute the Pearson correlation coefficient between the component number index of the input sample and the 1D outputSample */
-Point CorrelationAnalysis::PearsonCorrelation(const Sample & inputSample,
-    const Sample & outputSample)
+/**
+ * @class CorrelationAnalysis
+ */
+
+CLASSNAMEINIT(CorrelationAnalysis)
+
+/* Default constructor */
+CorrelationAnalysis::CorrelationAnalysis()
+  : PersistentObject()
+{
+  // nothing to do
+}
+
+/* Standard constructor */
+CorrelationAnalysis::CorrelationAnalysis(const Sample & inputSample,
+                                         const Sample & outputSample)
+  : PersistentObject()
+  , inputSample_(inputSample)
+  , outputSample_(outputSample)
 {
   if (outputSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
```

Comment:
Could we perform a single test here? i.e., `if (outputSample.getDimension() != 1 || inputSample.getDimension() != 1 ) throw InvalidDimensionException(HERE) << "Error: both samples must be 1D";`